File: programmers/python/level00/120835.py
# 진료순서 정하기
def solution(emergency):
    sortArr = sorted(list(emergency), reverse=True)
    return list(map(lambda num: sortArr.index(num) + 1, emergency))

# 다른 풀이
# def solution(emergency):
#     return [sorted(emergency, reverse=True).index(e) + 1 for e in emergency]

File: programmers/java/level00/120835.java
// 진료순서 정하기

import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;
import java.util.Comparator;

class Solution {
    public int[] solution(int[] emergency) {
        List<Integer> list = Arrays.stream(emergency)
            .boxed()
            .sorted(Comparator.reverseOrder())
            .collect(Collectors.toList());
        
        return Arrays.stream(emergency)
            .map(num -> list.indexOf(num) + 1)
            .toArray();
    }
}